src/rgw/rgw_rados.h

```diff
@@ -254,6 +254,14 @@ class RGWObjManifest {
                              as object might have been copied across buckets */
   map<uint64_t, RGWObjManifestRule> rules;
 
+  /* If this object is copied from another versioned object, @src_instance 
+   * is used to store src's version_id. 
+   * 
+   * @copied_obj is used to indicate whether @src_instance is valid.
+   * */
+  string src_instance; 
+  bool copied_obj;
```

Comment:
@yehudasa 
obj a (instance="", src_instance="") ---copy----> obj b (version_id="abc', src_instance="") ---copy---> obj c

without copied_obj，we can't decide obj c's src_instance's value. should it be obj b's src_instance (obj a's version_id) or obj b's version_id? Because empty src_instance is valid which should be copied into dest object's manifest in this senario.